### PR TITLE
Use less eval looking up value funcs

### PR DESCRIPTION
**Pull request by richo**
Technically this is a breaking chance, but with the old implementation
it's hard to do much usefully, so I'm willing to bet noone does

### CODE REVIEW

File: voltron/plugins/view/register.py [hunk 1]
```diff
@@ -367,10 +536,7 @@ def render(self):
                         if fmt['value_format'] != None and isinstance(formatted_reg, Number):
                             formatted_reg = fmt['value_format'].format(formatted_reg)
                         if fmt['value_func'] != None:
-                            if isinstance(fmt['value_func'], STRTYPES):
-                                formatted_reg = eval(fmt['value_func'])(formatted_reg)
-                            else:
-                                formatted_reg = fmt['value_func'](formatted_reg)
+                            formatted_reg = fmt['value_func'](self, formatted_reg)
                         if fmt['value_colour_en']:
```
Comment (richo) [inline voltron/plugins/view/register.py]: These should maybe something something elif? The last write wins behaviour is vaguely odd

**Comment by snare** [comment]
Did you test this? I'd be surprised if this works, since not all the register sections have a `format_func`, which is the reason for the "last write wins" behaviour.

**Comment by snare** [comment]
Oh wait that's accounted for.

**Comment by snare** [comment]
Reason for eval was originally that this would move to a config file, but it never happened. TBH I'd rather keep it the way it is and just replace the eval with a `getattr`

**Comment by snare** [comment]
`s/self.format_/format_/` and `formatted_reg = getattr(self, fmt['value_func'])(formatted_reg)`

File: voltron/plugins/view/register.py (hunk 1 above)
Comment (snare) [inline voltron/plugins/view/register.py]: No. The reason for this is that the register format config sections may have any combination of those fields.

**Comment by snare** [comment]
Fixed in 8e4ac9f893ace8bf48d1ecc17b449b3270fa01b3

**Pull request closed without merging**